### PR TITLE
DP-23372: Add the full URL as a field in the document CSV download file

### PR DESCRIPTION
**Pull request by arthurbaghdas**
**Description:**
Explain the technical implementation of the work done.

**Jira:** (Skip unless you are MA staff)
DP-23372

**To Test:**
- [ ] Add steps to test this feature

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: conf/drupal/config/views.view.all_documents.yml [hunk 1]
```diff
@@ -1351,23 +1351,24 @@ display:
...
           exclude: false
           alter:
-            alter_text: false
-            text: '<span title="{{ field_file_mime }}">{{ field_title }}</span>'
-            make_link: true
-            path: 'https://www.mass.gov/media/{{ mid }}/download'
-            absolute: false
-            external: true
+            alter_text: true
+            text: 'https://www.mass.gov{{ view_media_1 }}/download'
```
Comment (deviantintegral) [inline conf/drupal/config/views.view.all_documents.yml:1367]: I see this is hardcoded to the public URL. Should this be a relative link instead, so it works across different environments? Even so, it would probably be OK to do this as a new ticket, since this is unrelated and how it works on production today.

**Review by deviantintegral — COMMENTED** [review]
This looks close! Config changes look good (though I didn't do a line-by-line read). While I can see the new column, I'm not getting a downloaded file. Instead, after the batch completes I see the CSV in my browser (Safari). Given that's part of the ticket criteria, it probably needs to be fixed?

Screen Shot 2022-03-02 at 9 42 55 AM

File: conf/drupal/config/views.view.all_documents.yml (hunk 1 above)
Comment (arthurbaghdas) [reply conf/drupal/config/views.view.all_documents.yml:1367]: Yeah this is site specific. /download is a special URL in the site. So i think leaving it with hardcoded env URL is fine. At least we used same approach for different views as well.

**Comment by arthurbaghdas** [comment]
> This looks close! Config changes look good (though I didn't do a line-by-line read). While I can see the new column, I'm not getting a downloaded file. Instead, after the batch completes I see the CSV in my browser (Safari). Given that's part of the ticket criteria, it probably needs to be fixed?
> 
> Screen Shot 2022-03-02 at 9 42 55 AM

This issue is now fixed.

**Comment by deviantintegral** [comment]
Great! Process wise, should I be testing this since I reported the bug, or is there someone who will do cross-browser testing to make sure it didn't somehow break it elsewhere?

**Comment by arthurbaghdas** [comment]
> Great! Process wise, should I be testing this since I reported the bug, or is there someone who will do cross-browser testing to make sure it didn't somehow break it elsewhere?

**Comment by arthurbaghdas** [comment]
> Great! Process wise, should I be testing this since I reported the bug, or is there someone who will do cross-browser testing to make sure it didn't somehow break it elsewhere?

I think you can post this in the ticket.

**Review by beto-aveiga — APPROVED** [review]
LGTM
